### PR TITLE
ui: Admin UI Design Pass

### PR DESCRIPTION
**Pull request by mrtracy**
A number of commits working towards the Admin UI Refresh.

screen shot 2016-11-29 at 3 32 53 pm

---
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/11696)

### CODE REVIEW

**Comment by maxlang** [comment]
Minor note, but the Cockroach in the left nav doesn't have white wings anymore.

**Comment by mrtracy** [comment]
Hah, I did notice that; a problem, but that icon is not long for this world.

File: pkg/ui/app/containers/nodes.tsx [hunk 1]
```diff
@@ -1,40 +1,50 @@
...
+  { value: "resources", label: "System Resources" },
+  { value: "internals", label: "Advanced Internals" },
 ];
 
 /**
  * Renders the layout of the nodes page.
  */
 class ClusterOverview extends React.Component<ClusterOverviewProps, ClusterOverviewState> {
-  state = new ClusterOverviewState();
+  // Magic to add react router to the context.
```
Comment (maxlang) [inline pkg/ui/app/containers/nodes.tsx]: I know this is from my commit - I'm curious if you've looked into it yourself at all.

File: pkg/ui/app/components/summaryBar.tsx [hunk 2]
```diff
@@ -1,39 +1,61 @@
...
-  title: string;
-  tooltip: string;
+interface SummaryStatProps {
+  title: React.ReactNode;
+  tooltip?: string;
   value?: number;
   format?: (i: number) => string;
 }
 
+function computeValue(value: number, format?: (i: number) => string): string {
```
Comment (maxlang) [inline pkg/ui/app/components/summaryBar.tsx]: Could just make `_.indentity` the default and use format without checking.

File: pkg/ui/app/components/graphs.tsx [hunk 3]
```diff
@@ -104,7 +111,18 @@ class AxisDomain {
 
   // ticks computes 3 tick values for a graph given the current max/min.
```
Comment (maxlang) [inline pkg/ui/app/components/graphs.tsx]: Should probably fix comment to say it uses `this.tickcount`

File: pkg/ui/styl/base/palette.styl [hunk 4]
```diff
@@ -76,7 +76,16 @@ $secondary-gold-2 = #ECB51A // banner opt-in button hover
...
+// NEW PALETTE
+
+$dark-blue = #152849
+$light-green = #66AA26
+$light-blue = #5F6C87
+$glow-blue = #51BBFD
+$lighter-blue = #BEBECD
+$light-gray = #B8BBBC
+$light-gray-2 = #DCDCDC
+$lighter-gray = #F2F2F2
```
Comment (maxlang) [inline pkg/ui/styl/base/palette.styl]: Missing newline.

**Review by maxlang — APPROVED** [review]

**Comment by mrtracy** [comment]
Review status: 0 of 40 files reviewed at latest revision, 5 unresolved discussions, all commit checks successful.

---

*[pkg/ui/app/app.tsx, line 124 at r5](https://reviewable.io:443/reviews/cockroachdb/cockroach/11696#-KXlmq7SzzCHguHxKCrg-r5-124:-KXvY1bxu4HaNr5PwW_m:b-qhfj37) ([raw file](https://github.com/cockroachdb/cockroach/blob/ad5e4e3cf2276785b8913b95a0c2ac49e6bd0eab/pkg/ui/app/app.tsx#L124)):*
<details><summary><i>Previously, maxlang (Max Lang) wrote…</i></summary><blockquote>

I liked node being part of the query string because that will allow us to select multiple nodes, but we can always address this in the future if we need that functionality.
</blockquote></details>

Without multi-select, the data is completely hierarchical and fits well into URL path components.

What I mostly didn't like about this was how the dropdown control itself accessed the query string; I needed the same dropdown for the Databases page, but that was based on URL path components, and I would have had to modify those routes to use query strings instead if I wanted to re-use the dropdown navigation component. So, the really important change here was moving URL manipulation out of the dropdown component and up to the page itself.

---

*[pkg/ui/app/components/graphs.tsx, line 112 at r7](https://reviewable.io:443/reviews/cockroachdb/cockroach/11696#-KXlmq7VIJhFzpEWKreb-r7-112:-KXvYWWnHgzRxM_clOEa:b-y00g0o) ([raw file](https://github.com/cockroachdb/cockroach/blob/21af5a5a6ea2150e0fb4de608cb49f720702701e/pkg/ui/app/components/graphs.tsx#L112)):*
<details><summary><i>Previously, maxlang (Max Lang) wrote…</i></summary><blockquote>

Should probably fix comment to say it uses `this.tickcount`
</blockquote></details>

Will fix in subsequent PR.

---

*[pkg/ui/app/components/summaryBar.tsx, line 11 at r6](https://reviewable.io:443/reviews/cockroachdb/cockroach/11696#-KXlmq7TmLK81vdFKET--r6-11:-KXvYvaGFw9x5NTdMc2B:b-y00g0o) ([raw file](https://github.com/cockroachdb/cockroach/blob/44ed7bb9d2cf4bdc1602f60e870e255b0ca2ade3/pkg/ui/app/components/summaryBar.tsx#L11)):*
<details><summary><i>Previously, maxlang (Max Lang) wrote…</i></summary><blockquote>

Could just make `_.identity` the default and use format without checking.
</blockquote></details>

Will fix in subsequent PR.

---

*[pkg/ui/app/containers/nodes.tsx, line 38 at r5](https://reviewable.io:443/reviews/cockroachdb/cockroach/11696#-KXlmq7SzzCHguHxKCre-r5-38:-KXvYy0ZZz0O7WSSCErL:bjynuk) ([raw file](https://github.com/cockroachdb/cockroach/blob/ad5e4e3cf2276785b8913b95a0c2ac49e6bd0eab/pkg/ui/app/containers/nodes.tsx#L38)):*
<details><summary><i>Previously, maxlang (Max Lang) wrote…</i></summary><blockquote>

I know this is from my commit - I'm curious if you've looked into it yourself at all.
</blockquote></details>

Well, it's not really "magic", it's how react contexts are design. Some guy on that thread was complaining about it as "magic".

Anyways, this works fine, newer react-router versions provide a `withRouter()` method which is analogous to react-redux's `connect()` method - it's a HOC that provides the router via props.  We can move to using that at some point.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/11696)*

**Pull request merged**